apps/web/src/pages/Users/UpdateUserPage/components/TeamRoleTable.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { notEmpty } from "@gc-digital-talent/helpers";
+import { Heading, Link, Pill } from "@gc-digital-talent/ui";
+import { getLocalizedName } from "@gc-digital-talent/i18n";
+
+import Table, { ColumnsOf, Cell } from "~/components/Table/ClientManagedTable";
+import {
+  Role,
+  RoleAssignment,
+  UpdateUserAsAdminInput,
+  User,
+} from "~/api/generated";
+import useRoutes from "~/hooks/useRoutes";
+
+import AddTeamRoleDialog from "./AddTeamRoleDialog";
+import RemoveTeamRoleDialog from "./RemoveTeamRoleDialog";
+import { UpdateUserFunc } from "../types";
+
+const roleCell = (displayName: string) => {
+  return (
+    <Pill color="neutral" mode="solid">
+      {displayName}
+    </Pill>
+  );
+};
```

Comment:
This may be a better question for @Jerryescandon but, I beleive we want to list all roles associated with a specific team in a single row. Kind of like:

| Actions     | Team      | Roles                           |
| ----------- | --------- | ------------------------------- |
| edit/remove | Team Name | Pool Operator, Pool Admin, etc. |

Reply:
I think you're right, though it exposes a weird workflow.  If you want to remove a single team role while preserving a second team role on the same team you'll need to first remove all the team roles then add the preserved ones back in.  It's not important right now since we only have a single team role but it could get weird in the future.

Reply:
Yeah, I agree. We talked about this but I cannot remember the decision we came to 😢 

@tristan-orourke do you remember the outcome of that discussion we had the other week where we discussed this?

Reply:
The design for remove team membership looks like all or nothing to me.  If we roll up all the roles for a team to a single row, that row has a single "remove" button.
![image](https://user-images.githubusercontent.com/8978655/228273982-ccc38a3d-e4fe-4d0c-af4e-c66f052e221a.png)

Reply:
Well, you can click the "add new membership" button and select the team name in the list.  :laughing:

Reply:
OK, I've done the team rollup.  We'll probably need an "edit membership" dialog someday, but not today.  I'll send away for the updated translation.
ref: [roll up teams and fix EN copy](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6083/commits/2233ec4eaea6d1d02d70df77d7fbb714e37d4cde)

Reply:
Ah, I didn't realize the figma designs only had the delete button. In last week's discussion we decided there should be an edit button beside the delete button, just like on the current Team Membership table (using essentially the same dialogs) 
![image](https://user-images.githubusercontent.com/4313717/228343895-e1444ad4-67be-4f8f-b184-2c838ff93796.png)

Reply:
OK, edit dialog added in [Edit dialog](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6083/commits/40ad1e7b72fb9171cbf568a64f14b9143f79c7ec) and [tidy up edits](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6083/commits/002abbafbb9505a6f114fefc43700e8262851809)